mlrun/datastore/targets.py

```diff
@@ -554,7 +554,7 @@ def write_dataframe(
                 target_df,
                 self.storage_options or storage_options,
                 target_path,
-                partition_cols=partition_cols,
+                partition_cols=partition_cols or [],
```

Comment:
Pretty sure that's wrong. `partition_cols` can be `None`, and that has a different meaning from `[]`. You need to take a cue from the implementation that was there before #4216.